src/main.rs

```diff
@@ -32,27 +34,31 @@ fn main() -> Result<(), String> {
 }
 
 fn read_session_cookie(session_file: &Option<String>) -> String {
-    let path = if let Some(file) = session_file {
```

Comment:
I think the order of preference should be:
1. command line option (`--session-file`)
2. environment variable
3. `.adventofcode.session` in `~/.config`
4. `.adventofcode.session` in home directory